### PR TITLE
NAD-866 - Alert component review

### PR DESCRIPTION
**Pull request by marcelovicentegc**
#### What is the purpose of this pull request?

To update the current Alert component following the component review guidelines.

[Design specs are available on this Figma file.](https://www.figma.com/file/sepBhrHViUHyrTnktkuHaW/Admin-UI-%E2%80%93-Components?node-id=656%3A286721)

#### How should this be manually tested?

[Visit this storybook story ](https://alert-review.surge.sh/?path=/story/admin-ui-review-alert--tones)and its [documentation section](https://admin-ui-docs-git-feat-alert-review-admin-team.vercel.app/next/guidelines/components/alert). Make sure it matches the Figma specs.

### Breaking changes

- The `icon`, `fluid`, `visible`, and `sticky` properties were removed. 
- The `fluid` behavior was incorporated as default for it.
- Each variant has its own default icon now.
- The `tone` prop was renamed to `variant`.

You can use our codemod cli to update the alert's properties for you. [See its README for more information on how to execute the codemod](https://github.com/vtex/admin-ui/tree/main/packages/admin-ui-codemod#admin-ui-codemod).

#### Before
```tsx
import { Alert } from '@vtex/admin-ui'

function Example() {
  return (
    <Alert
      icon={<IconHeart />}
      fluid
      sticky
      visible
    >Hello world</Alert>
  )
}
```

#### After
```tsx
import { Alert } from '@vtex/admin-ui'

function Example() {
  return <Alert visible>Hello world</Alert>
}
```

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Quality improvement (tests or refactors)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [x] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui/src/alert/alert.test.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import { Alert } from './index'
+
+describe('Alert', () => {
+  beforeEach(jestMatchMedia)
+
+  it('should not have a11y violations', async () => {
+    const { container } = render(
+      <Fragment>
+        <Alert variant="positive">Order successfully placed</Alert>
```
Comment (matheusps) [inline packages/admin-ui/src/alert/alert.test.tsx:12]: Add an example of a visible alert

File: packages/admin-ui/src/alert/alert.style.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      critical: {
+        color: '$critical',
+        marginLeft: 'unset',
+      },
+      info: {
+        color: '$info',
+        marginLeft: 'unset',
+      },
+    }[variant]
+  )
```
Comment (matheusps) [inline packages/admin-ui/src/alert/alert.style.ts:70]:
```suggestion
export const iconContainerVariants = styleVariants({ 
  variant: {
    warning: {
      color: '$warning',
      marginLeft: 'unset',
    },
    positive: {
      color: '$positive',
      marginLeft: 'unset',
    },
    critical: {
      color: '$critical',
      marginLeft: 'unset',
    },
    info: {
      color: '$info',
      marginLeft: 'unset',
    },
  }
})
```

File: packages/admin-ui/src/alert/alert.style.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    return style({
+      marginLeft: '$2xl',
+      marginRight: '$xl',
+    })
+  }
+
+  return style({
+    marginLeft: '$2xl',
+  })
+}
```
Comment (matheusps) [inline packages/admin-ui/src/alert/alert.style.ts:83]:
```suggestion
export const buttonVariants = styleVariants({
  dismissible: {
    true: {
      marginLeft: '$2xl',
      marginRight: '$xl',
    },
    false: {
      marginLeft: '$2xl'
    }
  }
})
```

File: packages/admin-ui/src/alert/alert.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+const alertIconMap = {
+  critical: <IconXOctagon weight="fill" />,
+  positive: <IconCheckCircle weight="fill" />,
+  warning: <IconWarning weight="fill" />,
+  info: <IconBell weight="fill" />,
+}
+
+export type AlertActionProps = ButtonProps & {
+  children: ButtonProps['children']
+  onClick: ButtonProps['onClick']
```
Comment (matheusps) [inline packages/admin-ui/src/alert/alert.tsx:88]: These are already included on the ButtonProps, right?

File: packages/admin-ui/src/alert/alert.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+})
+
+const alertIconMap = {
+  critical: <IconXOctagon weight="fill" />,
+  positive: <IconCheckCircle weight="fill" />,
+  warning: <IconWarning weight="fill" />,
+  info: <IconBell weight="fill" />,
+}
+
+export type AlertActionProps = ButtonProps & {
```
Comment (matheusps) [inline packages/admin-ui/src/alert/alert.tsx:86]: You should nerf this type. Removing the props you already define internally (this is better for type documentation. So, omit: `variant`, `bleedX`, `bleedY`,  and `csx`.

Comment (matheusps) [reply packages/admin-ui/src/alert/alert.tsx:86]: Also, I don't think we need a spread here. The action API/footprint should be way smaller.

File: packages/admin-ui/src/alert/alert.test.tsx (hunk 1 above)
Comment (marcelovicentegc) [reply packages/admin-ui/src/alert/alert.test.tsx:12]: Alerts are now always visible

File: packages/admin-ui/src/alert/alert.style.ts (hunk 2 above)
Comment (marcelovicentegc) [reply packages/admin-ui/src/alert/alert.style.ts:70]: Holy cow how haven't I seen this?

File: packages/admin-ui/src/alert/alert.tsx (hunk 5 above)
Comment (marcelovicentegc) [reply packages/admin-ui/src/alert/alert.tsx:86]: Good catch!

**Review by matheusps — APPROVED** [review]

File: packages/admin-ui/src/alert/alert.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    children: (
+      <Fragment>
+        <Inline noWrap vSpace="unset" hSpace="$l">
+          <IconContainer size="regular" csx={styles.iconContainer({ variant })}>
+            {icon}
+          </IconContainer>
+          <Paragraph csx={styles.paragraph}>{children}</Paragraph>
+        </Inline>
+        <Inline noWrap vSpace="unset" hSpace="unset" csx={styles.rightInline}>
+          {action && (
```
Comment (Lenderson) [inline packages/admin-ui/src/alert/alert.tsx:47]: I would go with [ternary operator](https://pt-br.reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering): `action ? component : null `
Although, I think in [react 18 will be possible to return undefined](https://github.com/reactwg/react-18/discussions/75)

**Review by Lenderson — APPROVED** [review]

File: packages/admin-ui/src/alert/alert.tsx (hunk 6 above)
Comment (marcelovicentegc) [reply packages/admin-ui/src/alert/alert.tsx:47]: Good catch! Will do!

**Pull request merged**